### PR TITLE
146 deploy new conracts for testnet 2022 8 17

### PR DESCRIPTION
**Pull request by zh19910202**

### CODE REVIEW

File: deployments/rinkeby/P12Token.json [hunk 1]
```diff
@@ -0,0 +1,596 @@
+{
+  "address": "0x5cF89C84bF43E75F91b95807e118484E68fc4DC2",
```
Comment (Jerrywang959) [inline deployments/rinkeby/P12Token.json:2]: Set to exist p12 Token deployment

File: deploy/rinkeby/002_deploy_Uniswap_WETH.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  });
+
+  await deploy('UniswapV2Router02', {
+    from: deployer,
+    args: [(await get('UniswapV2Factory')).address, (await get('WETH9')).address],
+    log: true,
+  });
+};
+func.tags = ['Uniswap', 'WETH'];
+export default func;
```
Comment (Jerrywang959) [inline deploy/rinkeby/002_deploy_Uniswap_WETH.ts:27]: We don't need to deploy uniswap and weth on Uniswap

File: deployments/rinkeby/UniswapV2Factory.json [hunk 3]
```diff
@@ -0,0 +1,218 @@
+{
+  "address": "0x5c69bee701ef814a2b6a3edd4b1652cb9cc5aa6f",
```
Comment (Jerrywang959) [inline deployments/rinkeby/UniswapV2Factory.json:2]: use external deployed uniswap instead

File: deployments/p12TestNet/P12Token.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  "address": "0x75180e5F16A193fF8176A645bd9520b3eB522BC0",
+  "address": "0xeAc1F044C4b9B7069eF9F3eC05AC60Df76Fe6Cd0",
```
Comment (Jerrywang959) [inline deployments/p12TestNet/P12Token.json:2]: only address and abi are needed. Others could be removed as they are unrelavant

**Review by Jerrywang959 — APPROVED** [review]

**Review by Jerrywang959 — CHANGES_REQUESTED** [review]
approved accidentally

**Comment by zh19910202** [comment]
Useless information has been deleted

**Review by Jerrywang959 — APPROVED** [review]

**Pull request merged**